Playbook enhancements for Cloud CI

dargasudarshan opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
- This PR enhances the existing e2e framework to have a pre-condition check for cleanup activity of the previous test playbooks run failures.
- Also does a cleanup activity of /var/openebs in each minion to reclaim the disk space.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
@ksatchit @gprasath Can you please review this PR?

gprasath commented on e2e/ansible/pre-check.yml:
```diff
@@ -0,0 +1,55 @@
+
+  tasks:
+
+    - name: Cleanup /var/openebs
```
Can we use file module here to delete the directory?

dargasudarshan replied:
Modified task using file module.

gprasath commented on e2e/ansible/pre-check.yml:
```diff
@@ -0,0 +1,55 @@
+      args:
+        executable: /bin/bash
+      delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+      register: list_pv
```
can we remove all-namespaces here and proceed?

gprasath commented on e2e/ansible/pre-check.yml:
```diff
@@ -0,0 +1,55 @@
+      args:
+        executable: /bin/bash
+      delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+      with_items: "{{ list_pv.stdout_lines }}"
```
can we have a check which ensures the successful deletion?

dargasudarshan commented:
@gprasath Please find the playbook output here

```
vijay@vijay-virtual-machine:~/pr/openebs/e2e/ansible$ ansible-playbook pre-check.yml -vv
ansible-playbook 2.5.2
  config file = /home/vijay/pr/openebs/e2e/ansible/ansible.cfg
  configured module search path = [u'/home/vijay/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
Using /home/vijay/pr/openebs/e2e/ansible/ansible.cfg as config file

PLAYBOOK: pre-check.yml **************************************************************************************************************
1 plays in pre-check.yml

PLAY [localhost] *********************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************
task path: /home/vijay/pr/openebs/e2e/ansible/pre-check.yml:1
ok: [localhost]
META: ran handlers

TASK [Cleanup /var/openebs] **********************************************************************************************************
task path: /home/vijay/pr/openebs/e2e/ansible/pre-check.yml:8
changed: [localhost -> None] => (item=kubeminion01) => {"changed": true, "item": "kubeminion01", "path": "/var/openebs", "state": "absent"}
changed: [localhost -> None] => (item=kubeminion02) => {"changed": true, "item": "kubeminion02", "path": "/var/openebs", "state": "absent"}
changed: [localhost -> None] => (item=kubeminion03) => {"changed": true, "item": "kubeminion03", "path": "/var/openebs", "state": "absent"}

TASK [List available namespaces] *****************************************************************************************************
task path: /home/vijay/pr/openebs/e2e/ansible/pre-check.yml:16
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get ns --no-headers | awk {'print $1'}", "delta": "0:00:00.435810", "end": "2018-06-25 07:11:15.361992", "rc": 0, "start": "2018-06-25 07:11:14.926182", "stderr": "", "stderr_lines": [], "stdout": "default\ndrain-node\nkube-public\nkube-system\nopenebs\npercona-app\nreplica-quorum\nsingle-replica", "stdout_lines": ["default", "drain-node", "kube-public", "kube-system", "openebs", "percona-app", "replica-quorum", "single-replica"]}

TASK [Get list of pv's for all namespaces] *******************************************************************************************
task path: /home/vijay/pr/openebs/e2e/ansible/pre-check.yml:23
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get pv --no-headers | awk {'print $1'}", "delta": "0:00:00.372294", "end": "2018-06-25 07:11:16.132816", "rc": 0, "start": "2018-06-25 07:11:15.760522", "stderr": "", "stderr_lines": [], "stdout": "pvc-8b045540-75e3-11e8-9942-02b983f0a4db", "stdout_lines": ["pvc-8b045540-75e3-11e8-9942-02b983f0a4db"]}

TASK [Delete pv's] *******************************************************************************************************************
task path: /home/vijay/pr/openebs/e2e/ansible/pre-check.yml:30
changed: [localhost -> None] => (item=pvc-8b045540-75e3-11e8-9942-02b983f0a4db) => {"changed": true, "cmd": "source ~/.profile; kubectl delete pv pvc-8b045540-75e3-11e8-9942-02b983f0a4db", "delta": "0:00:00.222715", "end": "2018-06-25 07:11:16.728117", "item": "pvc-8b045540-75e3-11e8-9942-02b983f0a4db", "rc": 0, "start": "2018-06-25 07:11:16.505402", "stderr": "", "stderr_lines": [], "stdout": "persistentvolume \"pvc-8b045540-75e3-11e8-9942-02b983f0a4db\" deleted", "stdout_lines": ["persistentvolume \"pvc-8b045540-75e3-11e8-9942-02b983f0a4db\" deleted"]}

TASK [Delete namespaces which are not required] **************************************************************************************
task path: /home/vijay/pr/openebs/e2e/ansible/pre-check.yml:37
skipping: [localhost] => (item=default) => {"changed": false, "item": "default", "skip_reason": "Conditional result was False"}
changed: [localhost -> None] => (item=drain-node) => {"changed": true, "cmd": "source ~/.profile; kubectl delete ns drain-node --grace-period=0 --force", "delta": "0:00:00.186415", "end": "2018-06-25 07:11:17.285832", "item": "drain-node", "rc": 0, "start": "2018-06-25 07:11:17.099417", "stderr": "warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.", "stderr_lines": ["warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely."], "stdout": "namespace \"drain-node\" deleted", "stdout_lines": ["namespace \"drain-node\" deleted"]}
skipping: [localhost] => (item=kube-public) => {"changed": false, "item": "kube-public", "skip_reason": "Conditional result was False"}
skipping: [localhost] => (item=kube-system) => {"changed": false, "item": "kube-system", "skip_reason": "Conditional result was False"}
skipping: [localhost] => (item=openebs) => {"changed": false, "item": "openebs", "skip_reason": "Conditional result was False"}
changed: [localhost -> None] => (item=percona-app) => {"changed": true, "cmd": "source ~/.profile; kubectl delete ns percona-app --grace-period=0 --force", "delta": "0:00:00.193124", "end": "2018-06-25 07:11:17.763748", "item": "percona-app", "rc": 0, "start": "2018-06-25 07:11:17.570624", "stderr": "warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.", "stderr_lines": ["warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely."], "stdout": "namespace \"percona-app\" deleted", "stdout_lines": ["namespace \"percona-app\" deleted"]}
changed: [localhost -> None] => (item=replica-quorum) => {"changed": true, "cmd": "source ~/.profile; kubectl delete ns replica-quorum --grace-period=0 --force", "delta": "0:00:00.183624", "end": "2018-06-25 07:11:18.241974", "item": "replica-quorum", "rc": 0, "start": "2018-06-25 07:11:18.058350", "stderr": "warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.", "stderr_lines": ["warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely."], "stdout": "namespace \"replica-quorum\" deleted", "stdout_lines": ["namespace \"replica-quorum\" deleted"]}
changed: [localhost -> None] => (item=single-replica) => {"changed": true, "cmd": "source ~/.profile; kubectl delete ns single-replica --grace-period=0 --force", "delta": "0:00:00.178204", "end": "2018-06-25 07:11:18.669337", "item": "single-replica", "rc": 0, "start": "2018-06-25 07:11:18.491133", "stderr": "warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely.", "stderr_lines": ["warning: Immediate deletion does not wait for confirmation that the running resource has been terminated. The resource may continue to run on the cluster indefinitely."], "stdout": "namespace \"single-replica\" deleted", "stdout_lines": ["namespace \"single-replica\" deleted"]}

TASK [List namespaces after deletion] ************************************************************************************************
task path: /home/vijay/pr/openebs/e2e/ansible/pre-check.yml:46
FAILED - RETRYING: List namespaces after deletion (15 retries left).
changed: [localhost -> None] => {"attempts": 2, "changed": true, "cmd": "source ~/.profile; kubectl get ns --no-headers | awk {'print $1'}", "delta": "0:00:00.474154", "end": "2018-06-25 07:11:50.196269", "rc": 0, "start": "2018-06-25 07:11:49.722115", "stderr": "", "stderr_lines": [], "stdout": "default\nkube-public\nkube-system\nopenebs", "stdout_lines": ["default", "kube-public", "kube-system", "openebs"]}
META: ran handlers
META: ran handlers

PLAY RECAP ***************************************************************************************************************************
localhost                  : ok=7    changed=6    unreachable=0    failed=0
```
